### PR TITLE
Upgrade dependencies

### LINKED ISSUE #3: Update kotlinx-serialization
The old kotlinx-serialization version sadly causes a dependency conflict with [detekt-gitlab-report](https://gitlab.com/cromefire/detekt-gitlab-report) and because it's a built-in plugin it takes preference. It's working fine with 1.4.1 if constrained to it so upgrading should be fine.

### PR DESCRIPTION
**Pull request by cromefire**
Upgraded the following outdated dependencies (fixes #3 ):
- kotlin
- kotlinx-serialization
- dependencycheck
- junit (from 4 to junit 5/jupiter)
- gradle (via `:wrapper`)

### CODE REVIEW

**Review by chao2zhang — APPROVED** [review]

**Comment by chao2zhang** [comment]
Thank you for keeping our dependency up-to-date!

**Comment by cromefire** [comment]
I'm the author of [detekt-gitlab-report](https://gitlab.com/cromefire/detekt-gitlab-report), which has a conflict with the old kotlinx-serialization (and users currently have to use a work around to fix the newer version in gradle), so a full deployment would be great.

**Comment by chao2zhang** [comment]
0.2.0 should be available. Let me know whether the problem is resolved from your side

https://repo1.maven.org/maven2/io/github/detekt/sarif4k/sarif4k/0.2.0/

File: build.gradle.kts [hunk 1]
```diff
@@ -29,11 +32,26 @@ java {
...
+    withType<Javadoc> {
+        val customArgs = projectDir.resolve("javadoc-silence.txt")
+        customArgs.writeText(
+            """-Xdoclint:none
+            """.trimIndent()
+        )
+        options.optionFiles?.add(customArgs)
+    }
+
+    withType<Test> {
```
Comment (TWiStErRob) [inline build.gradle.kts:52]: As far as I remember this is eager, @chao2zhang, configureEach is the solution.
To validate, run `gradlew --scan`, in the Performance section it should show 0 (max 1 = :help) task created during configuration.

Comment (chao2zhang) [reply build.gradle.kts:52]: You are absolutely right. We could introduce https://github.com/autonomousapps/gradle-best-practices-plugin into not only this repo but also the main detekt repo.

This issue shouldn't impact the `sarif4k:0.2.0` though.

Comment (TWiStErRob) [reply build.gradle.kts:52]: I agree it doesn't directly affect this repo, but if you consider that this repo and Detekt is seen by a lot of people who use Gradle (e.g. Android devs) they might copy these patterns as "best practices", which is not good for any project.

I agree about checks to verify the builds, finally the community is trying to solve the "Big Gradle confusion", IntelliJ also added some static checks. I don't think a Gradle plugin is the best to detect these though, Detekt or other static analysis tools (probably PMD) might be better equipped. The plugin you mentioned has a good goal, but right now it's early stages: https://twitter.com/AutonomousApps/status/1571554472555581440

This is a way more comprehensive list of best practices to apply:
https://github.com/liutikas/gradle-best-practices

Comment (chao2zhang) [reply build.gradle.kts:52]: Fixed in https://github.com/detekt/sarif4k/commit/c6a5c5ea08d8965ab3a00595fcdcf42d64363bf4 🙏

**Pull request merged**